jetty-client/src/main/java/org/eclipse/jetty/client/util/InputStreamResponseListener.java

```diff
@@ -359,8 +359,11 @@ public void close() throws IOException
             if (LOG.isDebugEnabled())
                 LOG.debug("InputStream close");
 
-            Throwable failure = new AsynchronousCloseException();
-            callbacks.forEach(callback -> callback.failed(failure));
+            if (!callbacks.isEmpty())
+            {
+                Throwable failure = new AsynchronousCloseException();
+                callbacks.forEach(callback -> callback.failed(failure));
```

Comment:
Since this is performance, perhaps don't use for each? Actually we determined that streams are slower than a simple for loop, but does that include forEach ?